packages/grafana-ui/src/components/FileDropzone/FileDropzone.test.tsx

```diff
@@ -38,76 +38,60 @@ describe('The FileDropzone component', () => {
   });
 
   it('should handle file removal from the list', async () => {
-    const { rerender, container } = render(<FileDropzone />);
-    const dropzone = container.querySelector('div[TabIndex="0"]');
-
-    dispatchEvt(dropzone, 'drop', mockData(files));
+    render(<FileDropzone />);
 
-    await flushPromises(rerender, <FileDropzone />);
+    dispatchEvt(screen.getByTestId('dropzone'), 'drop', mockData(files));
 
-    expect(screen.getAllByLabelText('Remove')).toHaveLength(3);
+    expect(await screen.findAllByLabelText('Remove')).toHaveLength(3);
 
     fireEvent.click(screen.getAllByLabelText('Remove')[0]);
 
-    expect(screen.getAllByLabelText('Remove')).toHaveLength(2);
+    expect(await screen.findAllByLabelText('Remove')).toHaveLength(2);
   });
 
   it('should overwrite selected file when multiple false', async () => {
-    const component = <FileDropzone options={{ multiple: false }} />;
-    const { rerender, container } = render(component);
-    const dropzone = container.querySelector('div[TabIndex="0"]');
+    render(<FileDropzone options={{ multiple: false }} />);
 
-    dispatchEvt(dropzone, 'drop', mockData([file({})]));
-    await flushPromises(rerender, component);
+    dispatchEvt(screen.getByTestId('dropzone'), 'drop', mockData([file({})]));
 
-    expect(screen.getAllByLabelText('Remove')).toHaveLength(1);
-    expect(screen.getByText('ping.json')).toBeInTheDocument();
+    expect(await screen.findAllByLabelText('Remove')).toHaveLength(1);
+    expect(await screen.findByText('ping.json')).toBeInTheDocument();
```

Comment:
One `await` query is usually enough unless you're triggering another async action after it, the next ones can be `getBy`.

Reply:
I see, thanks!